cups/types/cups.py

```diff
@@ -0,0 +1,240 @@
+from dataclasses import (
+    dataclass,
+)
+from typing import (
+    Any,
+    ClassVar,
+    Dict,
+    Optional,
+)
+
+from cups.utils import (
+    _bytes_to_value,
+)
+
+from .base import cupsBaseClass, _ffi
+
+
+@dataclass(repr=False)
+class cupsOption(cupsBaseClass):
+    @property
+    def name(self) -> str:
+        return _bytes_to_value(self.ffi_value[0].name)
+
+    @property
+    def value(self) -> Optional[Any]:
+        return _bytes_to_value(self.ffi_value[0].value)
+
+    ffi_name = "cups_option_t"
+
+    def to_dict(self) -> dict:
+        return {"name": self.name, "value": self.value}
+
+    @classmethod
+    def from_cffi(cls, opt: Any) -> "cupsOption":
+        """
+        Convert a single CFFI dest struct to a Python cupsOption object.
+
+        Args:
+            dest (Any): The CFFI cups_option_t struct.
```

Comment:
The docstrings really help in this project, but are outdated, like the param name in this one.